plugins/utils/spectre.nix

```diff
@@ -0,0 +1,282 @@
+        )
+        ''
+          Keymaps declaration.
+
+          default: see [here](https://github.com/nvim-pack/nvim-spectre/blob/master/lua/spectre/config.lua)
+        '';
+
+      find_engine = mkEngineOption "find";
+
+      replace_engine = mkEngineOption "replace";
+
+      default = {
+        find = {
+          cmd = helpers.defaultNullOpts.mkStr "rg" ''
+            Which find engine to use. Pick one from the `find_engine` list.
+          '';
+
+          options = helpers.defaultNullOpts.mkListOf types.str ''["ignore-case"]'' ''
+            Options to use for this engine.
+          '';
+        };
+
+        replace = {
+          cmd = helpers.defaultNullOpts.mkStr "rg" ''
+            Which find engine to use. Pick one from the `replace_engine` list.
+          '';
+
+          options = helpers.defaultNullOpts.mkListOf types.str "[]" ''
+            Options to use for this engine.
+          '';
+        };
+      };
+
+      replace_vim_cmd = helpers.defaultNullOpts.mkStr "cdo" ''
+        The replace command to use within vim.
+      '';
+
+      is_open_target_win = helpers.defaultNullOpts.mkBool true ''
+        Open file on opener window.
+      '';
+
+      is_insert_mode = helpers.defaultNullOpts.mkBool false ''
+        Start open panel in insert mode.
+      '';
+
+      is_block_ui_break = helpers.defaultNullOpts.mkBool false ''
+        Mapping backspace and enter key to avoid ui break.
+      '';
+    };
+
+    settingsExample = {
+      live_update = true;
+      is_insert_mode = false;
+      find_engine = {
+        rg = {
+          cmd = "rg";
+          args = [
+            "--color=never"
+            "--no-heading"
+            "--with-filename"
+            "--line-number"
+            "--column"
+          ];
+          options = {
+            ignore-case = {
+              value = "--ignore-case";
+              icon = "[I]";
+              desc = "ignore case";
+            };
+            hidden = {
+              value = "--hidden";
+              desc = "hidden file";
+              icon = "[H]";
+            };
+            line = {
+              value = "-x";
+              icon = "[L]";
+              desc = "match in line";
+            };
+            word = {
+              value = "-w";
+              icon = "[W]";
+              desc = "match in word";
+            };
+          };
+        };
+      };
+      default = {
+        find = {
+          cmd = "rg";
+          options = ["word" "hidden"];
+        };
+        replace = {
+          cmd = "sed";
+        };
+      };
+    };
+
+    extraOptions = let
+      userCommandSettings = config.plugins.spectre.settings.default;
+    in {
+      findPackage = mkOption {
+        type = with types; nullOr package;
+        default =
+          {
+            rg = pkgs.ripgrep;
+          }
+          # `toString` will turn `null` into `"null"` to allow for the attrs indexation.
+          .${toString userCommandSettings.find.cmd}
```

Comment:
I wonder if it'd be more readable to have `{rg = pkgs.ripgrep;}` and `{sed = pkgs.gnused; inherit (pkgs) sd;}` defined in a let-block elsewhere. Perhaps alongside `userCommandSettings`?

Reply:
Is it fine for you now ? I agree that the readability is indeed better.